excluded_apps.txt

```diff
@@ -1,11 +1,17 @@
+64Gram.64Gram
+AnthonyBeaumont.AchievementWatcher
 Google.Chrome
+LogMeIn.Hamachi
 Microsoft.Edge
 Microsoft.EdgeWebView2Runtime
 Microsoft.Office
 Microsoft.OneDrive
+Microsoft.PowerShell
 Microsoft.Teams
+Microsoft.VisualStudioCode

```

Comment:
The thing is, with studio code for instance, that it can be install for system and require admin rights to update if I don't say wrong.

Reply:
Yes, but for now winget installs vscode with user rights and does not have a way to install as system https://github.com/microsoft/winget-cli/issues/965

Reply:
I use VS Code in system install, maintained by WAU: 
![image](https://user-images.githubusercontent.com/96626929/166142415-9474b617-0bb1-4939-ad54-54195816d50b.png)
and regarding https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/VisualStudioCode/1.66.2/Microsoft.VisualStudioCode.installer.yaml it has both installers. So for me, it does manage that properly. (It's purely my observations) 
But if VSCode is not installed, maybe on first install its ops for user install (not tested)

Reply:
I installed with winget running with admin rights, and it installed as user 🤔
I'll remove vscode from list for now.

Reply:
>and does not have a way to install as system

Oh, i totally missed `--scope` option and this https://github.com/microsoft/winget-cli/pull/819, by default winget will install as user to avoid UAC prompts

Reply:
I installed vscode now in windows terminal running as admin, and it chose user install.
There's two options:
1) Leave vscode excluded as most users will not (i guess?) use `--scope`
2) Remove vscode from excluded to allow unattended updates of vscode installed as system

Reply:
Don't know which one is best option. In all cases, it will update vscode in the scope it's already installed.
Regarding installation, maybe we need to force the machine scope, but don't know how it behaves if it is not supported in an app manifest.

Reply:
Anyway, something has to be done I think...
...using `& $Winget install --id` **without** `--scope machine` started under System context leads to ex. **Microsoft.VisualStudioCode** being installed to the System users own Profile, not %ProgramFiles%:
```
2022-05-05 03:05:38.903   -- File entry --
2022-05-05 03:05:38.903   Dest filename: C:\Windows\system32\config\systemprofile\AppData\Local\Programs\Microsoft VS Code\Code.exe
2022-05-05 03:05:38.903   Time stamp of our file: 2022-04-11 08:13:54.000
2022-05-05 03:05:38.903   Installing the file.
2022-05-05 03:05:42.418   Successfully installed the file.
```
That's not what we want, or?
I had the scope set mandatory to **machine** in my own previous solution, didn't notice this before now...

Reply:
I agree, as system context, we need to first try ti install with scope machine, and try to get the result. If failed due to scope, try to install without scope.
If an app need to be installed for user, with intune or SCCM easy to handle ;)

Reply:
Running as normal user:
```
PS $Username = "$env:Username"
PS $Username
knifmelti
PS whoami
knifmelti\knifmelti
PS $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
PS $ResolveWingetPath
PS
```
Running as admin user:
```
PS $Username = "$env:Username"
PS $Username
knifmelti
PS whoami
knifmelti\knifmelti
PS $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
PS $ResolveWingetPath
Path
----
C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10271.0_x64__8wekyb3d8bbwe
C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.18.431.0_x64__8wekyb3d8bbwe
```
Running as System:
```
PS $Username = "$env:Username"
PS $Username
WS-Desktop$
PS whoami
nt instans\system
PS $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
PS $ResolveWingetPath
Path
----
C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10271.0_x64__8wekyb3d8bbwe
C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.18.431.0_x64__8wekyb3d8bbwe
```